### PR TITLE
Small refactor: Include representations by composition

### PR DESCRIPTION
**Pull request by Prithvijutur**
Just moving around code mostly and removed also a couple of unused/deprecated public methods.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Prithvijutur has signed the CLA. Thanks for the pull request!

File: BoxContentSDK/BoxContentSDKTests/BOXFileRequestTests.m [hunk 1]
```diff
@@ -125,68 +126,64 @@ - (void)test_that_representation_options_are_set_correctly_when_request_is_perfo
     
     // Check x-reps-hint header original file request
     [fileRequest setRepresentationRequestOptions:@[@(BOXRepresentationRequestOriginal)]];
-    NSString *actualXRepsHint = [fileRequest formatRepresentationRequestHeader];
+    //NSString *actualXRepsHint = [fileRequest formatRepresentationRequestHeader];
```
Comment (Jeff-Meadows) [inline BoxContentSDK/BoxContentSDKTests/BOXFileRequestTests.m]: remove commented code line

File: BoxContentSDK/BoxContentSDK/Helper/BOXRepresentationsHelper.m [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+@end
+
+@implementation BOXRepresentationsHelper
+
+- (void)setRepresentationRequestOptions:(NSArray *)representationOptions
+{
+    self.representationsRequested = [[NSMutableOrderedSet alloc] initWithArray:representationOptions];
+}
+
+- (NSString *) getRepresentationsFieldString {
```
Comment (jdizoglio) [inline BoxContentSDK/BoxContentSDK/Helper/BOXRepresentationsHelper.m:22]: Should this be a class method?

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m [hunk 3]
```diff
@@ -16,12 +16,7 @@ @interface BOXFileRequest ()
 
 @property (nonatomic, readwrite, strong) NSString *fileID;
 @property (nonatomic, readwrite, assign) BOOL isTrashed;
-
-/**
- Setting a representation or list option will include availability of file with request representation information
- */
-@property (nonatomic, readwrite, strong) NSMutableOrderedSet *representationsRequested;
-
+@property (nonatomic, readonly, strong) BOXRepresentationsHelper* repsHelpers;
```
Comment (Jeff-Meadows) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m]: should this just be called `repsHelper`?

**Review by Jeff-Meadows — APPROVED** [review]

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m (hunk 3 above)
Comment (Prithvijutur) [reply BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m]: refresh:)

File: BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m [hunk 4]
```diff
@@ -16,12 +16,7 @@ @interface BOXFileRequest ()
 
 @property (nonatomic, readwrite, strong) NSString *fileID;
 @property (nonatomic, readwrite, assign) BOOL isTrashed;
-
-/**
- Setting a representation or list option will include availability of file with request representation information
- */
-@property (nonatomic, readwrite, strong) NSMutableOrderedSet *representationsRequested;
-
+@property (nonatomic, readonly, strong) BOXRepresentationsHelper* repsHelper;
```
Comment (ric0liva) [inline BoxContentSDK/BoxContentSDK/Requests/BOXFileRequest.m:19]: A helper as a property doesn't seem quite right.  Can we use a single-instance helper?

**Pull request merged**